editing/edit-features-with-feature-linked-annotation/src/main/java/com.esri.samples.edit_features_with_feature_linked_annotation/EditAttributesDialog.java

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright 2020 Esri.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.esri.samples.edit_features_with_feature_linked_annotation;
+
+import java.util.concurrent.ExecutionException;
+
+import javafx.fxml.FXML;
+import javafx.fxml.FXMLLoader;
+import javafx.scene.control.Alert;
+import javafx.scene.control.ButtonType;
+import javafx.scene.control.Dialog;
+import javafx.scene.control.TextField;
+
+import com.esri.arcgisruntime.data.Feature;
+import com.esri.arcgisruntime.concurrent.ListenableFuture;
+
+public class EditAttributesDialog extends Dialog<String>{
+
+  @FXML private TextField addressTextField;
+  @FXML private TextField streetNameTextField;
+  @FXML private ButtonType continueButton;
+
+  EditAttributesDialog(Feature selectedFeature){
+    FXMLLoader loader = new FXMLLoader(getClass().getResource("/edit_features_with_feature_linked_annotation/edit_attributes_dialog.fxml"));
+    loader.setRoot(this);
+    loader.setController(this);
+
+    setTitle("Edit Feature Attributes");
+
+    try {
+      loader.load();
```

Comment:
If I put the remaining code inside the `try` statement, when the loader fails, the rest of the code doesn't execute, but the code that executes after instantiating the dialog in `showEditableAttributes()` creates an empty dialog: 

![image](https://user-images.githubusercontent.com/67917108/89802718-c2689f80-db29-11ea-8558-8a8f598ce482.png)

However, if I remove the try catch entirely and have `EditAttributesDialog` throw the Exception, it can be handled in `showEditableAttributes()`:

```
try {
      // create a dialog to edit the attributes of the selected feature
      EditAttributesDialog editAttributesDialog = new EditAttributesDialog(selectedFeature);

      // show the dialog and wait for the user response
      editAttributesDialog.showAndWait();

      // update the selected feature's feature table
      updateAttributes(selectedFeature);
    } catch (Exception e) {
      new Alert(Alert.AlertType.ERROR, "Error: " + e).show();
    }
```
![image](https://user-images.githubusercontent.com/67917108/89802376-5be38180-db29-11ea-915d-c50f72662bb5.png)

Reply:
Good spot on these! 
> However, if I remove the try catch entirely and have EditAttributesDialog throw the Exception, it can be handled in showEditableAttributes():

I don't think it's a good idea to handle the exceptions in a different method, at least in the context of a sample. Best we keep the complexity to a minimum there.
So, looks like what you had originally was correct. Let's just leave it as is then 👍